### PR TITLE
Surreal 4

### PR DESCRIPTION
**Pull request by sctfn**
Some basic surreal work. I got a little stuck in the past few weeks, so I thought it would be best to send along what I've got. Easiest to review via file view.

### CODE REVIEW

**Comment by langgerard** [comment]
In the comment of sltn0, should not A and B be X and Y ?

File: set.mm [hunk 1]
```diff
@@ -533098,6 +533121,55 @@ property of surreals and will be used (via surreal cuts) to prove many
...
+      WGWLWIUTZSWNEIWEWGWAWMVRWDWAWFWBQRVRWDWFWMVAZSWGWLWIVBWNVQWLWFWHWELMVKVPV
+      QWDWFWMVDWPWQDEWHWEVEVFVGVHVI $.
+  $}
+
+  ${
+    $d A a b c $.  $d B a b c x $.  $d C a b c x y $.
+    $( If ` B ` is conitial with ` C ` and ` A ` precedes ` C ` , then ` A `
+       precedes ` B ` .  (Contributed by Scott Fenton, 24-Sep-2024.) $)
+    coinitsslt $p |- ( ( B e. ~P No /\
+     A. x e. B E. y e. C y <s x /\ A <<s C ) -> A <<s B ) $=
```
Comment (tirix) [inline set.mm:533162]: conitial > coinitial

**Review by tirix — APPROVED** [review]

File: set.mm [hunk 2]
```diff
@@ -532631,38 +532656,37 @@ C Fn ( ( S i^i dom F ) u. { z } ) ) $=
...
-      ( vx vy csslt wbr wss wa cvv wcel csur cslt wral w3a ssltex1 adantr simpr
-      cv ssexd ssltex2 ssltss1 sstrd ssltss2 ssltsep ssralv mpan9 3jca brsslt
-      syl21anbrc ) ABFGZCAHZIZCJKBJKZCLHZBLHZDSESMGEBNZDCNZOCBFGUMCAJUKAJKULABP
-      QUKULRZTUKUNULABUAQUMUOUPURUMCALUSUKALHULABUBQUCUKUPULABUDQUKUQDANULURDEA
-      BUEUQDCAUFUGUHDECBUIUJ $.
+      ( vx vy csslt wbr wss wa cvv wcel ssltex1 adantr simpr ssexd ssltex2 csur
+      ssltss1 sstrd cv ssltss2 cslt simp1l simp1r simp2 sseldd ssltsepc syl3anc
+      w3a simp3 ssltd ) ABFGZCAHZIZDECBJJUNCAJULAJKUMABLMULUMNZOULBJKUMABPMUNCA
+      QUOULAQHUMABRMSULBQHUMABUAMUNDTZCKZETZBKZUIZULUPAKUSUPURUBGULUMUQUSUCUTCA
+      UPULUMUQUSUDUNUQUSUEUFUNUQUSUJABUPURUGUHUK $.
```
Comment (icecream17) [inline set.mm:532663]: this proof gets longer

File: set.mm [hunk 3]
```diff
@@ -532834,23 +532858,22 @@ property of surreals and will be used (via surreal cuts) to prove many
...
-      cslt 3jca brsslt ) ACFGZBCFGZHZABUAZIJZCIJZHUPKLZCKLZDRERUJGECMZDUPMZUBUP
-      CFGUOUQURUMAIJBIJUQUNACNBCNABIIUCUDUMURUNACUEOUFUOUSUTVBUOABKUMAKLUNACPOU
-      NBKLUMBCPQUGUNUTUMBCUHQUOVADAMZVADBMZVBUMVCUNDEACSOUNVDUMDEBCSQVADABUITUK
-      DEUPCULT $.
+      ( vx vy csslt wbr cvv wcel ssltex1 adantr adantl csur ssltss1 cv ssltsepc
+      wss wi 3exp com12 wa cun unexd ssltex2 unssd ssltss2 cslt elun jaoi sylbi
+      wo 3imp21 ssltd ) ACFGZBCFGZUAZDEABUBZCHHUPABHHUNAHIUOACJKUOBHIUNBCJLUCUN
+      CHIUOACUDKUPABMUNAMQUOACNKUOBMQUNBCNLUEUNCMQUOACUFKDOZUQIZUPEOZCIZURUTUGG
+      ZUSURAIZURBIZUKUPVAVBRZRZURABUHVCVFVDUPVCVEUNVCVERUOUNVCVAVBACURUTPSKTUPV
+      DVEUOVDVERUNUOVDVAVBBCURUTPSLTUIUJULUM $.
```
Comment (icecream17) [inline set.mm:532866]: this proof gets longer

**Review by icecream17 — APPROVED** [review]

**Comment by sctfn** [comment]
> In the comment of sltn0, should not A and B be X and Y ?

Fixed

File: set.mm (hunk 1 above)
Comment (sctfn) [reply set.mm]: Fixed

File: set.mm (hunk 2 above)
Comment (sctfn) [reply set.mm]: New proof

File: set.mm (hunk 3 above)
Comment (sctfn) [reply set.mm:532866]: New proof

**Review by avekens — APPROVED** [review]

**Pull request merged**